packages/teams-js/src/internal/communication.ts

```diff
@@ -611,42 +621,52 @@ const handleParentMessageLogger = communicationLogger.extend('handleParentMessag
 function handleParentMessage(evt: DOMMessageEvent): void {
   const logger = handleParentMessageLogger;
 
-  if ('id' in evt.data && typeof evt.data.id === 'number') {
+  if ('id' in evt.data && (typeof evt.data.id === 'number' || typeof evt.data.id === 'string')) {
     // Call any associated Communication.callbacks
     const message = evt.data as MessageResponse;
-    const callback = CommunicationPrivate.callbacks[message.id];
-    logger('Received a response from parent for message %i', message.id);
+    const callbackId =
+      typeof message.id === 'string' ? message.id : CommunicationPrivate.legacyMessageIdsToUuidMap[message.id];
+    if (callbackId) {
+      validateUuid(callbackId);
```

Comment:
>validateUuid(callbackId); [](http://example.com/codeflow?start=6&length=25)

Rather than just validating the ID, can we instead use a UUID type that validates the data in its constructor and then throws if its not a real UUID?

Reply:
I really like this idea, amongst to the other ones suggested. I'll make those changes now